modules/csv/csv.go

```diff
@@ -102,3 +129,20 @@ func FormatError(err error, locale translation.Locale) (string, error) {
 
 	return "", err
 }
+
+// Looks for possible delimiters right before or after (with spaces after the former) double quotes with closing quotes
+var beforeAfterQuotes = regexp.MustCompile(`([,@\t;|]{0,1}) *(?:"[^"]*")+([,@\t;|]{0,1})`)
```

Comment:
So if I see that correctly, adding tabs between the separator (i.e. comma) and a quoted value, we will have an incorrectly guessed delimiter (tab)? I don't think we've seen the last of this issue with this approach.

Reply:
@delvh  Do people add tabs after a separator? I know many people find a comma delimited csv to be easier to read that way:

```
col1, col2, col3
a, b, c
```

But I can't image in doing 

```
col1, 	col2, 	col3
a, 	b, 	c
```

Are you suggesting to allow [\s]* instead of just a space so it gets all white space? Yet then I wonder if that could mess up when there are two tabs because no value for that cell 

```
col1 	col2 	col3
a 	 	c
```

Maybe 

```
`([,@\t;|]{0,1})[\s]*?(?:"[^"]*")+([,@\t;|]{0,1})`
```

as the regex?

Reply:
Just realized, the above regex I suggested would now fail on this, guessing comma instead of tab, even though the first row uses a tab:

```
col1	col2
he said,	"""hello"""
she said,	"""goodbye"""
```
Here the comma is part of the text. I think if this is the case, we should just use the guestDelimiter() function and not try to do this before/after quote as it might guess wrong. If we still use it, we should probably try it on the first row to make sure we get more than two columns returned, or maybe in guessDelimiter() we call guessFromBeforeAfterQuotes and we try that on the 10 lines first, rather than trying all delimiters and if it passes, just return it?

Reply:
I'm voting it is just best to go through the list of the delimiters and find what works for the sample 1 - 10 lines, removing all quoted strings so we have no values containing new lines. If we do this "find delimiter before/after quote" we still have to check if it works with the sample text.